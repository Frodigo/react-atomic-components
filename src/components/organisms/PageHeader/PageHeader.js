import React from 'react';
import PropTypes from 'prop-types';

import classes from './PageHeader.module.css';

const PageHeader = (props) => {
    const { logoUrl, logoAlt, linkUrl, navItems, ...rest } = props;

    const navLinks = navItems.map((link) => {
        return (
            <a
                key={link.label}
                href={link.url}
                className={`${link.isActive ? classes.navLinkActive : ''} ${
                    classes.navLink
                }`}
            >
                {link.label}
            </a>
        );
    });

    return (
        <header className={classes.pageHeader} {...rest}>
            <div className={classes.container}>
                <a href={linkUrl} className={classes.link}>
                   <img src={logoUrl} alt={logoAlt} />
                </a>

                <nav className={classes.primaryNav}>{navLinks}</nav>
            </div>
        </header>
    );
};

PageHeader.propTypes = {
    linkUrl: PropTypes.string.isRequired,
    navItems: PropTypes.arrayOf(
        PropTypes.shape({
            url: PropTypes.string.isRequired,
            label: PropTypes.string.isRequired,
            isActive: PropTypes.bool
        })
    )
};

PageHeader.defaultProps = {
    linkUrl: '/'
};

export default PageHeader;
