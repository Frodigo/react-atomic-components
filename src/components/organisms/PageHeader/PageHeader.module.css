.pageHeader {
    background-color: var(--color-secondary);
    padding: var(--spacing-md);
}

.container {
    align-items: center;
    display: flex;
    justify-content: center;
    height: 50px;
}


.link {
    display: inline-block;
    line-height: 0.8;
}

.primaryNav {
    display: none;
}

.navLink {
    color: var(--color-white);
    font-weight: 40;
    font-size: 14px;
    margin-left: var(--spacing-md);
    text-decoration: none;
}

.navLinkActive,
.navLink:hover,
.navLink:focus {
    text-decoration: underline;
}

@media (min-width: 1024px) {
    .pageHeader {

        padding: var(--spacing-md);
    }

    .container {
        justify-content: space-between;
    }

    .primaryNav {
        display: flex;
    }
}


@media (min-width: 1300px) {
    .container {
        margin: 0 auto;
        max-width: var(--desktop-container-width);
    }
}
