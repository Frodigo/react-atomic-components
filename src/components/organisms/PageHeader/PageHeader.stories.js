import React from 'react';

import { default as PageHeaderComponent } from './PageHeader';

export default {
    title: 'Organisms',
    component: PageHeaderComponent
};

const Template = (args) => <PageHeaderComponent {...args} />;

export const PageHeader = Template.bind({});
PageHeader.args = {
    navItems: [
        {
            url: '#',
            label: 'Home',
            isActive: true
        },
        {
            url: '#',
            label: 'About me'
        },
        {
            url: '#',
            label: 'Blog'
        },
        {
            url: '#',
            label: 'Get in touch'
        }
    ]
};
