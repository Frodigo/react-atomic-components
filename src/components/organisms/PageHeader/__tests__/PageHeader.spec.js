import React from 'react';
import { render } from '@testing-library/react';

import PageHeader from '../PageHeader';

describe('Page header should be rendered', () => {
    const pageHeaderNavItems = [
        {
            url: '#',
            label: 'Home'
        },
        {
            url: '#',
            label: 'About me',
            isActive: true
        },
        {
            url: '#',
            label: 'Blog'
        },
        {
            url: '#',
            label: 'Get in touch'
        }
    ];

    test('Page header renders correctly', () => {
        const { container } = render(
            <PageHeader navItems={pageHeaderNavItems} />
        );

        expect(container).toMatchSnapshot();
    });
});
