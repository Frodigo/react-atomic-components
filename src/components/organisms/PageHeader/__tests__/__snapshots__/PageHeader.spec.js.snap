// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page header should be rendered Page header renders correctly 1`] = `
<div>
  <header
    class="pageHeader"
  >
    <div
      class="container"
    >
      <a
        class="link"
        href="/"
      >
        <img />
      </a>
      <nav
        class="primaryNav"
      >
        <a
          class=" navLink"
          href="#"
        >
          Home
        </a>
        <a
          class="navLinkActive navLink"
          href="#"
        >
          About me
        </a>
        <a
          class=" navLink"
          href="#"
        >
          Blog
        </a>
        <a
          class=" navLink"
          href="#"
        >
          Get in touch
        </a>
      </nav>
    </div>
  </header>
</div>
`;
